File: SDKs/amazonaws.com/eks/2017-11-01/go/pkg/models/shared/updateparamtypeenum.go
package shared

type UpdateParamTypeEnum string

const (
	UpdateParamTypeEnumVersion                  UpdateParamTypeEnum = "Version"
	UpdateParamTypeEnumPlatformVersion          UpdateParamTypeEnum = "PlatformVersion"
	UpdateParamTypeEnumEndpointPrivateAccess    UpdateParamTypeEnum = "EndpointPrivateAccess"
	UpdateParamTypeEnumEndpointPublicAccess     UpdateParamTypeEnum = "EndpointPublicAccess"
	UpdateParamTypeEnumClusterLogging           UpdateParamTypeEnum = "ClusterLogging"
	UpdateParamTypeEnumDesiredSize              UpdateParamTypeEnum = "DesiredSize"
	UpdateParamTypeEnumLabelsToAdd              UpdateParamTypeEnum = "LabelsToAdd"
	UpdateParamTypeEnumLabelsToRemove           UpdateParamTypeEnum = "LabelsToRemove"
	UpdateParamTypeEnumTaintsToAdd              UpdateParamTypeEnum = "TaintsToAdd"
	UpdateParamTypeEnumTaintsToRemove           UpdateParamTypeEnum = "TaintsToRemove"
	UpdateParamTypeEnumMaxSize                  UpdateParamTypeEnum = "MaxSize"
	UpdateParamTypeEnumMinSize                  UpdateParamTypeEnum = "MinSize"
	UpdateParamTypeEnumReleaseVersion           UpdateParamTypeEnum = "ReleaseVersion"
	UpdateParamTypeEnumPublicAccessCidrs        UpdateParamTypeEnum = "PublicAccessCidrs"
	UpdateParamTypeEnumLaunchTemplateName       UpdateParamTypeEnum = "LaunchTemplateName"
	UpdateParamTypeEnumLaunchTemplateVersion    UpdateParamTypeEnum = "LaunchTemplateVersion"
	UpdateParamTypeEnumIdentityProviderConfig   UpdateParamTypeEnum = "IdentityProviderConfig"
	UpdateParamTypeEnumEncryptionConfig         UpdateParamTypeEnum = "EncryptionConfig"
	UpdateParamTypeEnumAddonVersion             UpdateParamTypeEnum = "AddonVersion"
	UpdateParamTypeEnumServiceAccountRoleArn    UpdateParamTypeEnum = "ServiceAccountRoleArn"
	UpdateParamTypeEnumResolveConflicts         UpdateParamTypeEnum = "ResolveConflicts"
	UpdateParamTypeEnumMaxUnavailable           UpdateParamTypeEnum = "MaxUnavailable"
	UpdateParamTypeEnumMaxUnavailablePercentage UpdateParamTypeEnum = "MaxUnavailablePercentage"
)

File: SDKs/amazonaws.com/eks/2017-11-01/python/src/sdk/models/shared/updateparamtype_enum.py
from dataclasses import dataclass, field
from typing import Enum

class UpdateParamTypeEnum(str, Enum):
    VERSION = "Version"
    PLATFORM_VERSION = "PlatformVersion"
    ENDPOINT_PRIVATE_ACCESS = "EndpointPrivateAccess"
    ENDPOINT_PUBLIC_ACCESS = "EndpointPublicAccess"
    CLUSTER_LOGGING = "ClusterLogging"
    DESIRED_SIZE = "DesiredSize"
    LABELS_TO_ADD = "LabelsToAdd"
    LABELS_TO_REMOVE = "LabelsToRemove"
    TAINTS_TO_ADD = "TaintsToAdd"
    TAINTS_TO_REMOVE = "TaintsToRemove"
    MAX_SIZE = "MaxSize"
    MIN_SIZE = "MinSize"
    RELEASE_VERSION = "ReleaseVersion"
    PUBLIC_ACCESS_CIDRS = "PublicAccessCidrs"
    LAUNCH_TEMPLATE_NAME = "LaunchTemplateName"
    LAUNCH_TEMPLATE_VERSION = "LaunchTemplateVersion"
    IDENTITY_PROVIDER_CONFIG = "IdentityProviderConfig"
    ENCRYPTION_CONFIG = "EncryptionConfig"
    ADDON_VERSION = "AddonVersion"
    SERVICE_ACCOUNT_ROLE_ARN = "ServiceAccountRoleArn"
    RESOLVE_CONFLICTS = "ResolveConflicts"
    MAX_UNAVAILABLE = "MaxUnavailable"
    MAX_UNAVAILABLE_PERCENTAGE = "MaxUnavailablePercentage"

